### PR TITLE
Remove deprecation warning for Swift 4.2

### PR DESCRIPTION
**Pull request by yhkaplan**
This appeared to be the only warning that came up when using Swift 4.2/Xcode 10, other than a couple warnings from dependencies that haven't been updated yet.

Also, this updates the travis config to use Swift 4.2/Xcode 10

### CODE REVIEW

**Comment by yhkaplan** [comment]
Hmm, looks like swiftenv on linux might not have 4.2 yet...

`The command "swiftenv global 4.2" failed and exited with 1 during .`

Also, Xcode 10 requires a fairly recent version of macOS (10.13.6?), but the version on Travis is too old...

```
ProductName:	Mac OS X
ProductVersion:	10.13.3
```

**Comment by yhkaplan** [comment]
4.2 final release version not available yet for swiftenv 😿 
```sh
❯ swiftenv install --list
#...
4.1-dev
4.1
4.2-dev
5.0-dev
```

File: Sources/Danger/File.swift [hunk 1]
```diff
@@ -19,13 +19,8 @@ extension File {
...
 
-public enum FileType: String, Equatable {
+public enum FileType: String, Equatable, CaseIterable {
     case h, json, m, markdown = "md", mm, pbxproj, plist, storyboard, swift, xcscheme, yaml, yml
-
-    @available(swift, deprecated: 4.2, message: "Replace with CaseIterable conformance")
-    static internal var allCases: [FileType] {
-        return [.h, .json, .m, .markdown, .mm, .pbxproj, .plist, .storyboard, .swift, .xcscheme, yaml, yml]
-    }
 }
```
Comment (orta) [inline Sources/Danger/File.swift]: Any chance you can make this a backwards compatible change?

Comment (SD10) [reply Sources/Danger/File.swift]: @yhkaplan You should be able to wrap the property in `#if !(swift(>=4.2))`. 

Btw I noticed the `yaml` and `yml` cases in the array aren't prefixed with a dot... isn't that odd... or am I missing something?

Comment (yhkaplan) [reply Sources/Danger/File.swift]: > yaml and yml cases in the array aren't prefixed with a dot

Really odd that the compiler didn't catch this... Will fix!

Comment (yhkaplan) [reply Sources/Danger/File.swift]: I've added backwards compatibility (and added the missing periods) in the commit below. The only thing I'm wavering about still is what to set the CI to. (Also it looks like I forgot to set the .swift-version file)

I could set the macOS and Linux versions to Swift 4.2 and 4.1 respectively, but what do you think? Or we could wait to merge this until Swift 4.2 is available on Swiftenv, which may be soon.

Comment (yhkaplan) [reply Sources/Danger/File.swift]: It looks like Swift 4.2 support was added to Swiftenv in the master branch on Sep 19, just needs to be released

https://github.com/kylef/swiftenv/commit/a949ff6ab8d1386f1bc99219da1ba860ea7de35a

Comment (yhkaplan) [reply Sources/Danger/File.swift]: The install script for Swiftenv picks up the latest version which contains the Swift 4.2 installation link, so the failure point on Linux and macOS just seems to be Danger-JS right now.

```
Error:  TypeError: Cannot read property 'concat' of undefined
    at Object.exports.mergeResults
```

**Comment by erikstrottmann** [comment]
I’m getting Swift Package Manager failures with danger 4.0.2 and danger-swift 0.4.1 when importing a plugin with Marathon. In my case, I’m importing [danger-swiftlint](https://github.com/ashfurrow/danger-swiftlint) 0.3.0, but it doesn’t appear to be particular to that plugin.

It looks like [Marathon 3.1.0](https://github.com/JohnSundell/Marathon/releases/tag/3.1.0) adds support for Swift Package Manager 4.2. It’s also backward-compatible with earlier versions of the Swift compiler.

Bottom line, danger-swift needs to list Marathon 3.1.0 (instead of 3.0.0) in its `Package.swift` to fully support the Swift 4.2 compiler.

**Comment by yhkaplan** [comment]
@erikstrottmann Thanks for the heads up! I’ll update Marathon in this PR today or this weekend.

**Comment by yhkaplan** [comment]
@erikstrottmann I'm unable to build both danger-swift with Marathon 3.1 or marathon w/ Xcode 9.4.x (Swift 4.1). Instead, only Xcode 10/Swift 4.2 is working for me, so this appears to break backward compatibility. Were you able to get Marathon 3.1 building in Xcode 9.4?

**Comment by orta** [comment]
if we don't have a choice, I'm fine with moving forwards

**Comment by erikstrottmann** [comment]
I’m able to build both danger-swift depending on Marathon 3.1.0, and Marathon 3.1.0 itself using both Xcode 9.4 / Swift 4.1 and Xcode 10 / Swift 4.2. But I’m not sure how to test my locally built copy of danger-swift. It looks like danger-js is defaulting to the version of danger-swift that I installed using Homebrew, which works with Swift 4.1 but fails with Swift 4.2:

```
~/Developer/ios (master=) $ DANGER_GITHUB_API_TOKEN="<redacted>" DANGER_FAKE_CI="YEP" DANGER_TEST_REPO="<redacted>" pr="<redacted>" /opt/boxen/nodenv/shims/danger process ../danger-swift/.build/debug/danger-swift
Dangerfile.swift:2:8: error: module compiled with Swift 4.1.2 cannot be imported by the Swift 4.2 compiler: /usr/local/lib/danger/Danger.swiftmodule
impo
rt Danger
       ^

$ cd "/Users/erikstrottmann/.danger-swift/Scripts/Cache/-Users-erikstrottmann-Developer-ios-_dangerfile_imports/" && readlink OriginalFile
/Users/erikstrottmann/Developer/ios/_dangerfile_imports.swift
$ cd "/Users/erikstrottmann/Developer/ios/" && /usr/bin/env xcrun --sdk macosx swift package --version
Apple Swift Package Manager - Swift 4.2.0 (swiftpm-14248)
$ cd "/Users/erikstrottmann/Developer/ios/" && /usr/bin/env xcrun --sdk macosx swift package --version
Apple Swift Package Manager - Swift 4.2.0 (swiftpm-14248)
$ cd "/Users/erikstrottmann/.danger-swift/Scripts/Cache/-Users-erikstrottmann-Developer-ios-_dangerfile_imports/" && /usr/bin/env xcrun --sdk macosx swift build -C /Users/erikstrottmann/.danger-swift/Scripts/Cache/-Users-erikstrottmann-Developer-ios-_dangerfile_imports/ 
Compile Swift Module 'Wrap' (1 sources)
Compile Swift Module 'ShellOut' (1 sources)
Compile Swift Module 'Require' (1 sources)
Compile Swift Module 'Files' (1 sources)
Compile Swift Module 'Danger' (11 sources)
Compile Swift Module 'Unbox' (42 sources)
Compile Swift Module 'Releases' (4 sources)
Compile Swift Module 'MarathonCore' (29 sources)
Linking ./.build/x86_64-apple-macosx10.10/debug/libDanger.dylib
Compile Swift Module 'Marathon' (1 sources)
Compile Swift Module 'Runner' (5 sources)
Compile Swift Module 'DangerSwiftLint' (3 sources)
Linking ./.build/x86_64-apple-macosx10.10/debug/Marathon
Linking ./.build/x86_64-apple-macosx10.10/debug/danger-swift
Compile Swift Module '_dangerfile_imports' (1 sources)
Linking ./.build/x86_64-apple-macosx10.10/debug/libMarathonDependencies.dylib
Linking ./.build/x86_64-apple-macosx10.10/debug/_dangerfile_imports
Running: /usr/bin/swiftc --driver-mode=swift -L /usr/local/lib/danger -I /usr/local/lib/danger -lDanger -L /usr/local/lib/danger -I /usr/local/lib/danger -L /Users/erikstrottmann/.danger-swift/Scripts/Cache/-Users-erikstrottmann-Developer-ios-_dangerfile_imports/.build/debug -I /Users/erikstrottmann/.danger-swift/Scripts/Cache/-Users-erikstrottmann-Developer-ios-_dangerfile_imports/.build/debug -lMarathonDependencies Dangerfile.swift /var/folders/xj/xcb77bdj0d96t1jhwdpgqwy00000gn/T/danger-dsl.json /var/folders/xj/xcb77bdj0d96t1jhwdpgqwy00000gn/T/danger-response.json
ERROR: Dangerfile eval failed at Dangerfile.swift
ERROR: Could not get the results JSON file at /var/folders/xj/xcb77bdj0d96t1jhwdpgqwy00000gn/T/danger-response.json

Error:  TypeError: Cannot read property 'concat' of undefined
    at Object.exports.mergeResults (/opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/reporting.js:15:33)
    at Object.<anonymous> (/opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/runDangerSubprocess.js:110:51)
    at step (/opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/runDangerSubprocess.js:32:23)
    at Object.next (/opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/runDangerSubprocess.js:13:53)
    at /opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/runDangerSubprocess.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/runDangerSubprocess.js:3:12)
    at ChildProcess.<anonymous> (/opt/nodes/9.8.0/lib/node_modules/danger/distribution/commands/utils/runDangerSubprocess.js:98:48)
    at ChildProcess.emit (events.js:180:13)
    at maybeClose (internal/child_process.js:936:16)
```

**Comment by orta** [comment]
👍

**Comment by orta** [comment]
Let's do this then.
